src/fileioc/fileioc.h

```diff
@@ -382,6 +382,38 @@ uint8_t ti_RclVar(const uint8_t var_type, const char *var_name, void **data_stru
  */
 bool ti_ArchiveHasRoom(uint24_t num_bytes);
 
+
+/**
+ * Set routine to run after a garbage collect.
+ * @param routine Routine to run following a garbage collect. NULL sets it to do nothing.
+ * @note If your program uses graphx, pass gfx_Begin to setup graphics after the garbage collect finishes.
+ * */
+void ti_SetPostGCHandler(void (*routine)(void));
+
+/**
+ * Set routine to run before a garbage collect.
+ * @param routine Routine to run preceeding a garbage collect. NULL sets it to do nothing. If this routine returns true, the variable will not be archived.
```

Comment:
It should probably also be mentioned that, if the routine returns `true`, the post-GC handler will not be run.

Reply:
Actually, should I set the post-gc handler to run regardless of a gc triggering?

Reply:
I was thinking about that, but if the pre-GC handler returns `true`, it makes sense to just do that before making any changes in preparation for a GC that's not going to happen that would need to be reverted by the post-GC handler.

Reply:
Fair enough.